### PR TITLE
Feature/issue 31

### PR DESCRIPTION
**Pull request by NishantNaithani1997**
Created a single parser file which can be used by other modules

### CODE REVIEW

File: expressEngine/lib/parser/parser.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export default (controller = null, functionName = '') => async (req, res) => {
```
Comment (nodexpertsdev) [inline expressEngine/lib/parser/parser.js]: @NishantNaithani1997 Parser is not acting as a middleware here, please update it, use a json to validate all fields on the basis of request. Pass only required field to controller using next().

Comment (nodexpertsdev) [inline expressEngine/lib/parser/parser.js]: @NishantNaithani1997 Parser is not acting as a middleware here, please update it, use a json to validate all fields on the basis of request. Pass only required field to controller using next().

Comment (kartikay-nx) [inline expressEngine/lib/parser/parser.js]: @NishantNaithani1997 Parser is not acting as a middleware here, please update it, use a json to validate all fields on the basis of request. Pass only required field to controller using next().

**Comment by NishantNaithani1997** [comment]
Link for more reference: https://github.com/nodexpertsdev/RnD/issues/31

File: expressEngine/lib/parser/parser.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const { params, query, body } = req;
+
+  try {
+    if (!(controller && controller[functionName])) {
+      throw new Error(`No function found in controller having name : ${functionName}`);
```
Comment (mayank-nx) [inline expressEngine/lib/parser/parser.js]: @NishantNaithani1997 Please move this text to cms

**Comment by mayank-nx** [comment]
@NishantNaithani1997 Please make sure to use parser as middleware.

**Pull request merged**